enhancements/two-nodes-openshift/2no.md

```diff
@@ -439,38 +494,51 @@ See template for guidelines/instructions.
 - What impact do these API extensions have on existing SLIs (e.g. scalability, API throughput,
   API availability)
 
-  [TODO: Expand] Toggling CEO control values with result in etcd being briefly offline.
+  Toggling CEO control values with result in etcd being briefly offline. The transition is almost immediate, though, since the resource agent is watching for the
+  etcd pod to disappear so it can start its replacement.
+
+  The other potential impact is around reboots. There may be a small performance impact when the nodes reboot since they have to leave the etcd cluster and resync etcd to join.
 
 - How is the impact on existing SLIs to be measured and when (e.g. every release by QE, or
   automatically in CI) and by whom (e.g. perf team; name the responsible person and let them review
   this enhancement)
 
+  The impact of the etcd transition as well as the reboot interactions with etcd are likely compatible with existing SLIs.
+
 - Describe the possible failure modes of the API extensions.
 
+  There shouldn't be any failures introduced by adding a new topology.
+  Etcd transitioning from CEO-managed to externally managed should be a one-way switch. If it fails, it should result in a failed installation.
+  Similarly, introducing an operator to remediate the fencing agent exists so that we can degrade the operator if the fencing credentials are invalid or if fencing cannot be initialized,
+  leading to a failed installation. The special case is if the BMC credentials are rotated. Here, the operator's role is to notify the cluster admin that the pacemaker fencing agent is unhealthy.
+
+  If the administrator fails to remediate this before a fencing operation is required, then manual recovery will be required by SSH-ing to the node.
+
 - Describe how a failure or behaviour of the extension will impact the overall cluster health
   (e.g. which kube-controller-manager functionality will stop working), especially regarding
   stability, availability, performance, and security.
+
+  As mentioned above, a BMC credential rotation could result in a customer "breaking" pacemaker's ability to fence nodes. On its own, pacemaker has no way of communicating this kind
+  of failure to the cluster admin. The proposed 2NO resource agent remediation operator could detect this and warn the cluster administrator that action is required. Alternatively,
+  RHEL-HA could detect this and immediately respond by turning off etcd for one of the nodes - forcing the cluster into read-only mode until a user manually logs in and restores the fencing agent.
```

Comment:
btw, OpenShift has no concept of "read-only". Since etcd and apiserver requires quorum for both writes and (linearized) reads, it will never return anything and your cluster will be down. 

Workload still continues to run.

Reply:
This part needs to be updated based on today's discussion since pacemaker don't have an active role in this. it just let's an already broken cluster stay broken because it can proceed with fencing.